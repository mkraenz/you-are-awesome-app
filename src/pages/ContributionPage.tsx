import React, { Component } from "react";
import { Alert, View } from "react-native";
import { Header, Icon } from "react-native-elements";
import { HeaderTitle } from "react-navigation-stack";
import { connect } from "react-redux";
import { Dispatch } from "redux";
import uuid from "uuid";
import { IPostAdded } from "../redux/Actions";
import { IPost, IPostContent } from "../redux/IPost";
import { IReduxState } from "../redux/IReduxState";
import { ReduxAction } from "../redux/ReduxAction";
import AddPostInput from "./AddPostInputs";
import { INavigationProps } from "./INavigationProps";
import { styles } from "./Styles";

interface Props extends INavigationProps {
    addPost: (post: IPost) => void;
}

class ContributionPage extends Component<Props> {
    public render() {
        return (
            <View style={styles.container}>
                <Header
                    leftComponent={
                        <Icon
                            name="arrow-back"
                            onPress={() => this.props.navigation.goBack()}
                            color={styles.header.color}
                        />
                    }
                    centerComponent={
                        <HeaderTitle style={styles.header}>
                            Contribute
                        </HeaderTitle>
                    }
                ></Header>
                <AddPostInput
                    handleSubmit={(post: IPostContent) =>
                        this.handleSubmit(post)
                    }
                />
            </View>
        );
    }

    private handleSubmit(post: IPostContent) {
        this.props.addPost({
            ...post,
            id: uuid.v4(),
        });
        Alert.alert(
            "Thanks for your contribution!",
            `You are a valued member of our awesome community. \n Your message: \n ${post.text}`,
            [
                {
                    text: "Awesome!",
                    onPress: () => this.props.navigation.navigate("Home"),
                },
            ]
        );
    }
}

const mapStateToProps = (state: IReduxState) => state;

const mapDispatchToProps = (dispatch: Dispatch) => ({
    addPost: (payload: IPost): IPostAdded =>
        dispatch({
            type: ReduxAction.PostAdded,
            payload,
        }),
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(ContributionPage);
