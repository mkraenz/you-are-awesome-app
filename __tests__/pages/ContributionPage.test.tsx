import React from "react";
import "react-native";
import { Provider } from "react-redux";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import { default as createMockStore } from "redux-mock-store";
import ContributionPage from "../../src/pages/ContributionPage";

it("renders correctly", () => {
    const store = createMockStore([])({
        netInfo: {
            connected: true,
        },
    });

    const tree = renderer
        .create(
            <Provider store={store}>
                <ContributionPage />
            </Provider>
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

it("renders no connection notify if disconnected from internet", () => {
    const store = createMockStore([])({
        netInfo: {
            connected: false,
        },
    });

    const tree = renderer
        .create(
            <Provider store={store}>
                <ContributionPage />
            </Provider>
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
    // TODO expect No internet connection
});
