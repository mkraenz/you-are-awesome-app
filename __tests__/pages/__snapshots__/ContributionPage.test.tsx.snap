// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#89cff0",
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2089dc",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingTop": 20,
      }
    }
  >
    <Image
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 64,
            "width": undefined,
          },
          undefined,
        ]
      }
      testID="headerContainer"
    />
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }
    >
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
        }
      }
    >
      <Text
        accessibilityTraits="header"
        numberOfLines={1}
        style={
          Object {
            "color": "white",
            "fontSize": 17,
            "fontWeight": "600",
            "marginHorizontal": 16,
          }
        }
      >
        Contribute
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "paddingTop": 23,
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "dimgrey",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginTop": 10,
          }
        }
      >
        Awesome message
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="sentences"
          editable={true}
          maxLength={200}
          onChangeText={[Function]}
          placeholder="Your awesome message to the world"
          placeholderTextColor="grey"
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "center",
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "dimgrey",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginTop": 10,
          }
        }
      >
        First Name
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="words"
          editable={true}
          maxLength={50}
          onChangeText={[Function]}
          placeholder="First name"
          placeholderTextColor="grey"
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "center",
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "dimgrey",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginTop": 10,
          }
        }
      >
        Country
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="sentences"
          editable={true}
          maxLength={50}
          onChangeText={[Function]}
          placeholder="Country"
          placeholderTextColor="grey"
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "center",
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#2289dc",
          "height": 40,
          "margin": 15,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "textAlign": "center",
          }
        }
      >
         Submit 
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders no connection notify if disconnected from internet 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#89cff0",
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2089dc",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingTop": 20,
      }
    }
  >
    <Image
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 64,
            "width": undefined,
          },
          undefined,
        ]
      }
      testID="headerContainer"
    />
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }
    >
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
        }
      }
    >
      <Text
        accessibilityTraits="header"
        numberOfLines={1}
        style={
          Object {
            "color": "white",
            "fontSize": 17,
            "fontWeight": "600",
            "marginHorizontal": 16,
          }
        }
      >
        Contribute
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#b52424",
        "flexDirection": "row",
        "height": 30,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
        }
      }
    >
      No Internet Connection
    </Text>
  </View>
  <View
    style={
      Object {
        "paddingTop": 23,
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "dimgrey",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginTop": 10,
          }
        }
      >
        Awesome message
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="sentences"
          editable={true}
          maxLength={200}
          onChangeText={[Function]}
          placeholder="Your awesome message to the world"
          placeholderTextColor="grey"
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "center",
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "dimgrey",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginTop": 10,
          }
        }
      >
        First Name
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="words"
          editable={true}
          maxLength={50}
          onChangeText={[Function]}
          placeholder="First name"
          placeholderTextColor="grey"
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "center",
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "dimgrey",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginTop": 10,
          }
        }
      >
        Country
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="sentences"
          editable={true}
          maxLength={50}
          onChangeText={[Function]}
          placeholder="Country"
          placeholderTextColor="grey"
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "center",
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#2289dc",
          "height": 40,
          "margin": 15,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "textAlign": "center",
          }
        }
      >
         Submit 
      </Text>
    </View>
  </View>
</View>
`;
